### PR TITLE
Refactor to Next.js 13 and `app` directory

### PR DESCRIPTION
**Pull request by gfortaine**

### CODE REVIEW

File: components/comment.js [hunk 1]
```diff
@@ -1,82 +1,75 @@
...
+            />,
+            <div key="children" className="children">
+              {comments.map((comment) => (
+                <Comment key={comment.id} {...comment} />
+              ))}
+            </div>,
+          ]}
+
+      <style jsx>{`
+        .comment {
```
Comment (leerob) [inline components/comment.js:40]: Let's change these to CSS Modules?

File: app/ssr/page.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { experimental_use as use } from 'react'
+
+import Page from '../../components/page.client'
+import Story from '../../components/story.client'
```
Comment (leerob) [inline app/ssr/page.js:4]: We don't need the `.client` extension, just `use client`

Comment (leerob) [inline app/ssr/page.js:4]: Can we move all of the components into the `app` directory to colocate?

File: app/layout.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export default function RootLayout({ children }) {
+    return (
+      <html>
+        <head />
```
Comment (leerob) [inline app/layout.js:4]: Can we add some meta tags like title here?

File: components/comment.js (hunk 1 above)
Comment (gfortaine) [reply components/comment.js:40]: Unfortunately, it looks like that CSS Modules don't work with Client Components. I am preparing a PR with some failing tests that should have passed.

Comment (gfortaine) [reply components/comment.js]: https://github.com/vercel/next.js/issues/42072

File: app/ssr/page.js (hunk 2 above)
Comment (gfortaine) [reply app/ssr/page.js]: What is the point of moving all of the components into the `app` directory ?

https://github.com/vercel/next.js/tree/canary/examples/layout-component

Comment (leerob) [reply app/ssr/page.js]: We don't have to but it's nice to highlight you can colocate things now.

**Comment by leerob** [comment]
Hey! I don't have push access here. I made a PR: https://github.com/gfortaine/next-react-server-components/pull/1

File: app/layout.js (hunk 3 above)
Comment (gfortaine) [reply app/layout.js]: It looks like that meta is also required to be able to support CSS media queries : https://github.com/vercel/next.js/issues/41842#issuecomment-1292104770

**Comment by gfortaine** [comment]
@leerob PR merged 🎉

**Comment by leerob** [comment]
Hey! Sorry one thing I should have been more clear about - I think we can ditch all of the CSR/slow examples and just do the RSC/streaming one as the main index page.

**Comment by gfortaine** [comment]
@leerob Copy that 👍 MR updated accordingly  :

- bugfixes :
  - use react@^18.3.0-next (fix html mismatch errors between client & server + parentNode not found from loading.js)
  - remove `fetchData` in `components/item.js`
  - remove delay from `fetchData` function call in `app/page.js`
- CSS modules everywhere possible
- various code refactoring

**Comment by vercel-team** [comment]
#### 📝 Changed routes:




6 deleted routes:
- **/csr**
- **/**
- **/item.server**
- **/rsc.server**
- **/slow.server**
- **/ssr**

Commit 7972519a9c32438624403ab628bb6dbb54fb882a (https://next-rsc-news-6z85w4785.vercel.sh).

**Comment by gfortaine** [comment]
@leerob Just discovered a very annoying bug by finishing this MR (to get the `id` searchParam in `ItemPage` Server Component) : https://github.com/vercel/next.js/issues/42856

**Comment by leerob** [comment]
Thank you so much for your help here! Sending you an email for some swag 😄 This PR is already getting fairly large, so let's go ahead and merge forward and address bugs as we see them.

For example, I think there should be no surrounding padding on mobile. But it's minor, and good that this repo gets updated so folks don't try exploring the old version that won't work 😄

**Review by leerob — APPROVED** [review]

**Comment by mb21** [comment]
Hello! any particular reason you're using a plain old html `<a>` tag instead of the [Next `<Link` component](https://beta.nextjs.org/docs/api-reference/components/link)? This is essentially a MPA. From reading the beta docs for app directory, I understood the recommended approach to still be SPA? Thanks for clarification!

**Pull request merged**